src/perfetto_mcp/connection_manager.py

```
@@ -74,6 +92,11 @@ def _create_connection(self, trace_path: str) -> TraceProcessor:
             )
         except Exception as e:
             logger.error(f"Failed to connect to trace: {trace_path}, error: {e}")
+            if self._trace_processor_bin_path:
+                raise ConnectionError(
+                    "Could not connect to trace processor with configured "
+                    f"{TRACE_PROCESSOR_BIN_PATH_ENV}={self._trace_processor_bin_path}: {e}"
+                )
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)